proposals/z-index.md

```diff
@@ -0,0 +1,56 @@
+# z-index Convention
+
+
+## Problem
+We have no convention for choosing z-index. This means that developers can set an arbitrary z-index on a component that may have the side effect that it displays above or below another component of which it has no knowledge.
+
+
+## Solution
+
+### Base layers
+First we define the layers in an object.
+
+```js
+const layers = {
```

Comment:
I think this is a great starting point. Requiring getting But using numbers as keys doesn't provide any context.

I would like to see all the keys provide some sort of context such as in all these examples:
https://css-tricks.com/handling-z-index/
https://www.smashingmagazine.com/2014/06/sassy-z-index-management-for-complex-layouts/
https://www.sitepoint.com/better-solution-managing-z-index-sass/

This will require more though to be put in up front, but will provide a more scalable solution long term.

Reply:
I agree on this point.